runtime/descriptor.go

```diff
@@ -117,5 +119,31 @@ func makeStructFieldDescriptor(t *Type, fieldName, propertyName string) *Object
 		}
 		return ret, raised
 	}
+	if writtable {
+		setterFunc := func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+			ret := None
+			var raised *BaseException
+			if raised = checkFunctionArgs(f, fieldName, args, ObjectType, ObjectType); raised == nil {
```

Comment:
Prefer to [return early](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow) rather than having nested if statements:

```
if raised := checkFunctionArgs(...); raised != nil {
  return nil, raised
}
```